### DISCUSSION TITLE
请问能在 Xray(M+F+H+K+A)+Caddy(N) 的配置文件中实现 Vless Vision TLS 回落到 Reality 吗？

### DISCUSSION THREAD

**Discussion by jliveusa** [lxhao61/integrated-examples#209] [Q&A] [answered]
如上，谢谢大佬！

**Comment by lxhao61** [comment, marked as answer]
你想错了，XTLS Vision 回落只能到 HTTP/1.1 及 H2C 网站， 不能回落到 REALITY (HTTP/2 网站)。
另外 REALITY Vision 应用是 XTLS Vision 应用的改进（可使用外部证书与强制客户端使用 uTLS 指纹），无必要同时使用。
 若还是需要同时使用，参考 Xray(E+F+H+A)+Caddy(N) 在 Xray(M+F+H+K+A)+Caddy(N) 中分别增加 SNI 分流对应部分与 XTLS Vision 应用部分即可。

**Comment by jliveusa** [comment]
再次请教，服务器上有两个网站，a网站为正常网站，路径为www/a，b网站为科学网站www/b，如何使用caddy进行分流？尝试过好几次都失败了omg

**Reply by lxhao61** [reply]
若你直接使用上面示例来实现，正常网站或科学网站随你 SNI 分流指定即可。

**Reply by jliveusa** [reply]
"sni": {
          "listen": [":443"],
          "routes": [{
            "match": [{
              "tls": {
                "sni": ["xxxx.com"] 
            "handle": [{
              "handler": "headers",
              "response": {
                "set": {
                  "Strict-Transport-Security": ["max-age=31536000; includeSubDomains; preload"] 
                }
              }
            },
            {
              "handler": "file_server",
              "root": "/var/www/html" 
            }]
          }],
          },
          {
            "match": [{
              "tls": {
                "sni": ["trojan.xxxxx.com"] 
              }
            }],
            "handle": [{
              "handler": "proxy",
              "upstreams": [{
                "dial": ["unix/@trojan.sock"]
              }],
              "proxy_protocol": "v2" 
            }]
          },

尝试过这种模式去实现，但是caddy报错了，烦请帮忙看一下，谢谢~

**Reply by lxhao61** [reply]
肯定报错！你站点应用配置怎么在 SNI 分流配置块中配置？
